kernel/src/syscall/rename.rs

```diff
@@ -11,6 +11,25 @@ use crate::{
     syscall::constants::MAX_FILENAME_LEN,
 };
 
+pub fn sys_renameat2(
+    old_dirfd: FileDesc,
+    old_path_addr: Vaddr,
+    new_dirfd: FileDesc,
+    new_path_addr: Vaddr,
+    flags: u32,
+    ctx: &Context,
+) -> Result<SyscallReturn> {
+    let Some(flags) = Flags::from_bits(flags) else {
+        return_errno_with_message!(Errno::EINVAL, "invalid flags");
+    };
+    // TODO: support the handling of `NOREPLACE`, `EXCHANGE`, and `WHITEOUT` flags.
+    if !flags.is_empty() {
+        warn!("unsupported flags: {:?}", flags);
+        return_errno_with_message!(Errno::EINVAL, "unsupported flags");
+    }
+    sys_renameat(old_dirfd, old_path_addr, new_dirfd, new_path_addr, ctx)
+}
+
```

Comment:
Since `sys_renameat2` is a superset of `sys_renameat`, you should let sys_renameat call sys_renameat2, instead of reverse.

```rust
pub fn sys_renameat2(
    old_dirfd: FileDesc,
    old_path_addr: Vaddr,
    new_dirfd: FileDesc,
    new_path_addr: Vaddr,
    flags: u32,
    ctx: &Context,
) -> Result<SyscallReturn> {
     /// the details.......
}

pub fn sys_renameat(
    old_dirfd: FileDesc,
    old_path_addr: Vaddr,
    new_dirfd: FileDesc,
    new_path_addr: Vaddr,
    ctx: &Context,
) -> Result<SyscallReturn> {
     sys_renameat2(old_dirfd,
        old_path_addr,
        new_dirfd,
        new_path_add,
        0,
        ctx)
}


```